### DISCUSSION TITLE
adding pnpm project to a new pnpm monorepo

### DISCUSSION THREAD

**Discussion by michaelchiche** [pnpm/pnpm#5579] [Q&A]
Hi,

I am having issues trying to add a repository/package into a new monorepo using pnpm.
This is the structure I have for my project
```
├── apps
│   ├── docs
│   └── demo-app <─ Folder that i am adding
├── package.json
├── packages
│   ├── css
│   └── components
├── pnpm-lock.yaml
└── pnpm-workspace.yaml
```

`demo-app` already has a `pnpm-lock.yaml`, but when I do a `pnpm install` at the root of the monorepo, it installs dependencies using what is specified in the `package.json` and not the pinned versions from the `pnpm-lock.yaml`. 

Is this expected ? and is there a way to use the pinned dependencies from the `pnpm-lock.yaml` in `demo-app` in the monorepo (other that specifying the deps in the `package.json`

**Comment by lvqq** [comment]
~~You can try `--frozen-lockfile` option: https://pnpm.io/cli/install#--frozen-lockfile~~

**Reply by michaelchiche** [reply]
Hi @lvqq ,

Thank you for your answer, but I dont understand how this helps ?
The problem is that the monorepo does not install the pinned version in the `pnpm-lock.yaml` from one the packages.

From what I understand, this command says to not generate a lockfile

**Reply by lvqq** [reply]
Sorry, I misunderstand your question. For monorepo, the lock file won't work if it's not in the root directory by default.  The config `shared-workspace-lockfile = false` in `.npmrc` can help it: https://pnpm.io/npmrc#shared-workspace-lockfile
